### PR TITLE
Jk signup

### PR DESCRIPTION
**Pull request by jenkens-dev**

### CODE REVIEW

**Comment by jenkens-dev** [comment]
@alphy11

File: src/components/SignUpDialogueBox.tsx [hunk 1]
```diff
@@ -1,24 +1,89 @@
...
+const REGISTER_VOLUNTEER = gql`
+  mutation registerVolunteer(
+    $username: String!
+    $password: String!
+    $confirmPassword: String!
+  ) {
+    registerVolunteer(
+      username: $username
+      password: $password
+      confirmPassword: $confirmPassword
```
Comment (Alphy11) [inline src/components/SignUpDialogueBox.tsx:24]: I think for password confirmation, I would do that check client side.  If someone is signing in through the API they can accidentally make a typo in their password :-)

File: src/components/SignUpDialogueBox.tsx [hunk 2]
```diff
@@ -1,24 +1,89 @@
...
+    ) {
+      ok
```
Comment (Alphy11) [inline src/components/SignUpDialogueBox.tsx:26]: Is it normal to request these things back?  Doesn't graphql have some protocol in place for errors? (I  have no idea, I just would have assumed)

File: src/components/SignUpDialogueBox.tsx [hunk 3]
```diff
@@ -1,24 +1,89 @@
...
 // Header, Text Fields, Radio Selection, Submit
 const SignUpDialogueBox = (props: DialogueProps) => {
-  const [username, setUsername] = useState("");
-  const [password, setPassword] = useState("");
-  const [passwordConfirm, setPasswordConfirm] = useState("");
-  const [mission, setMission] = useState("");
-  const [description, setDescription] = useState("");
-  const [displayName, setDisplayName] = useState("");
-  const [contact, setContact] = useState("");
+  const [username, setUsername] = useState('');
```
Comment (Alphy11) [inline src/components/SignUpDialogueBox.tsx:74]: There's lots of form libraries out there that will do a lot of this for you.

File: src/components/SignUpDialogueBox.tsx [hunk 4]
```diff
@@ -1,24 +1,89 @@
...
+  const [confirmPassword, setConfirmPassword] = useState('');
+  const [mission, setMission] = useState('');
+  const [description, setDescription] = useState('');
+  const [displayName, setDisplayName] = useState('');
+  const [contact, setContact] = useState('');
+
+  const client = useApolloClient();
+  let history = useHistory();
+
+  const [registerVolunteer] = useMutation(REGISTER_VOLUNTEER);
```
Comment (Alphy11) [inline src/components/SignUpDialogueBox.tsx:85]: I'd split the volunteer and nonprofit into two components with their own submit behaviors, you've got a lot more `if( volunteer)` than things that are not different.

File: src/components/SignUpDialogueBox.tsx [hunk 5]
```diff
@@ -55,18 +120,53 @@ const SignUpDialogueBox = (props: DialogueProps) => {
...
+        },
+      });
+      if (nonprofitData.data.registerNonprofit.ok) {
+        client.writeData({
+          data: {
+            nonprofitID: nonprofitData.data.registerNonprofit.nonprofit.id,
+            token: nonprofitData.data.registerNonprofit.token,
+          },
+        });
+        history.push('/nonprofit-dashboard');
```
Comment (Alphy11) [inline src/components/SignUpDialogueBox.tsx:161]: would recommend looking into ReactRouterV5.1 it's nicer to use.  Not high priority though

File: src/components/SignUpDialogueBox.tsx [hunk 6]
```diff
@@ -55,18 +120,53 @@ const SignUpDialogueBox = (props: DialogueProps) => {
...
+      }
+    }
   };
 
   return (
-    <div className="bg-blue-500 flex flex-col h-full px-4 pt-4">
+    <form
+      className="bg-blue-500 flex flex-col h-full px-4 pt-4"
```
Comment (Alphy11) [inline src/components/SignUpDialogueBox.tsx:168]: css classes? GROSS

**Review by Alphy11 — APPROVED** [review]
Nothing wrong, just some ideas for how it could be organized a bit better

File: src/components/SignUpDialogueBox.tsx (hunk 2 above)
Comment (jenkens-dev) [reply src/components/SignUpDialogueBox.tsx:26]: I did this for some easier to read error messages that we can display. Like validation stuff. Errors in [useMutation ](https://www.apollographql.com/docs/react/data/mutations/) hook confuse me. In the useQuery hook I have access to errors.

File: src/components/SignUpDialogueBox.tsx (hunk 3 above)
Comment (jenkens-dev) [reply src/components/SignUpDialogueBox.tsx:74]: Have you used Formik? I'm looking at their [example](https://codesandbox.io/s/8y578o8152) form in TS and I'm confused how they are able to update their state values

File: src/components/SignUpDialogueBox.tsx (hunk 6 above)
Comment (jenkens-dev) [reply src/components/SignUpDialogueBox.tsx:168]: not actually css classes this is just Tailwind syntax

File: src/components/SignUpDialogueBox.tsx (hunk 4 above)
Comment (jenkens-dev) [reply src/components/SignUpDialogueBox.tsx:85]: this confuses me

**Pull request merged**